os/kernel/pthread/pthread_keycreate.c

```diff
@@ -97,8 +97,9 @@ int pthread_key_create(FAR pthread_key_t *key, pthread_destructor_t destructor)
 
 	DEBUGASSERT(group);
 
+	DEBUGASSERT(key);
```

Comment:
If the group is null, it should not be happened. When it's null, then it means something wrong. That's why we should call assert here.
But for key, I think just error return is enough.